chain/client/src/sync/epoch.rs

```diff
@@ -191,8 +224,14 @@ impl EpochSync {
                 Error::Other("Could not find merkle proof for first block".to_string())
             })?;
 
+        let all_past_epochs_including_old_proof = existing_epoch_sync_proof
+            .map(|proof| proof.past_epochs)
+            .unwrap_or_else(Vec::new)
+            .into_iter()
+            .chain(all_past_epochs.into_iter())
```

Comment:
Had a slight confusion for a moment, forgot that these are `all_past_epochs` AFTER `last_epoch_we_have_proof_for`.
Maybe rename it to sth like `all_past_epochs_since_last_proof` ?
And then we could use `all_past_epochs` instead of `all_past_epochs_including_old_proof` ?